modules/core/include/opencv2/core/quaternion.inl.hpp

```diff
@@ -866,6 +866,194 @@ Quat<T> Quat<T>::spline(const Quat<T> &q0, const Quat<T> &q1, const Quat<T> &q2,
     return squad(vec[1], s1, s2, vec[2], t, assumeUnit, QUAT_ASSUME_NOT_UNIT);
 }
 
+namespace detail {
+
+template <typename T> static
+Quat<T> createFromAxisRot(int axis, const T theta)
+{
+    if (axis == 0)
+        return Quat<T>::createFromXRot(theta);
+    if (axis == 1)
+        return Quat<T>::createFromYRot(theta);
+    if (axis == 2)
+        return Quat<T>::createFromZRot(theta);
+    CV_Assert(0);
+}
+
+template <typename T> static
+bool isIntAngleType(typename QuatEnum::EulerAnglesType eulerAnglesType)
```

Comment:
Please drop template from here. It is not needed anymore with `QuatEnum`

Reply:
OK!